### PR TITLE
Add checksum for py-ipywidgets 8.0.2, py-jupyterlab-widgets 3.0.3

### PR DESCRIPTION
**Pull request by iarspider**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-jupyterlab-widgets/package.py [hunk 1]
```diff
@@ -26,6 +31,7 @@ class PyJupyterlabWidgets(PythonPackage):
     )
 
     depends_on("python@3.6:", type=("build", "run"))
+    depends_on("python@3.7:", when="@3.0.3:", type=("build", "run"))
     depends_on("py-setuptools@40.8.0:", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-jupyterlab-widgets/package.py]: If we're installing from a wheel, I think you can remove all of these build-only dependencies

File: var/spack/repos/builtin/packages/py-widgetsnbextension/package.py [hunk 2]
```diff
@@ -11,14 +11,19 @@ class PyWidgetsnbextension(PythonPackage):
...
     version("3.5.1", sha256="079f87d87270bce047512400efd70238820751a11d2d8cb137a5a5bdbaf255c7")
     version("3.4.2", sha256="fa618be8435447a017fd1bf2c7ae922d0428056cfc7449f7a8641edf76b48265")
     version("3.4.0", sha256="c9d6e426a1d79d132b57b93b368feba2c66eb7b0fd34bdb901716b4b88e94497")
     version("3.3.0", sha256="c5280a62d293735cdadc7b8884e2affcfb0488420ee09963577f042359726392")
     version("1.2.6", sha256="c618cfb32978c9517caf0b4ef3aec312f8dd138577745e7b0d4abfcc7315ce51")
 
-    depends_on("py-setuptools", type="build")
+    depends_on("py-setuptools", when="@:3", type="build")
+    # TODO: replace this after concretizer learns how to concretize separate build deps
+    depends_on("py-jupyter-packaging10", when="@4.0.3:", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-widgetsnbextension/package.py]: I don't think you need to add a new package for this, `py-jupyter-packaging11` will satisfy `~=0.10`

File: var/spack/repos/builtin/packages/py-widgetsnbextension/package.py [hunk 3]
```diff
@@ -11,14 +11,19 @@ class PyWidgetsnbextension(PythonPackage):
...
+    version("4.0.3", sha256="34824864c062b0b3030ad78210db5ae6a3960dfb61d5b27562d6631774de0286")
     version("3.6.0", sha256="e84a7a9fcb9baf3d57106e184a7389a8f8eb935bf741a5eb9d60aa18cc029a80")
     version("3.5.1", sha256="079f87d87270bce047512400efd70238820751a11d2d8cb137a5a5bdbaf255c7")
     version("3.4.2", sha256="fa618be8435447a017fd1bf2c7ae922d0428056cfc7449f7a8641edf76b48265")
     version("3.4.0", sha256="c9d6e426a1d79d132b57b93b368feba2c66eb7b0fd34bdb901716b4b88e94497")
     version("3.3.0", sha256="c5280a62d293735cdadc7b8884e2affcfb0488420ee09963577f042359726392")
     version("1.2.6", sha256="c618cfb32978c9517caf0b4ef3aec312f8dd138577745e7b0d4abfcc7315ce51")
 
-    depends_on("py-setuptools", type="build")
+    depends_on("py-setuptools", when="@:3", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-widgetsnbextension/package.py]: Setuptools is still needed I believe

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-widgetsnbextension/package.py (hunk 3 above)
Comment (iarspider) [reply var/spack/repos/builtin/packages/py-widgetsnbextension/package.py]: I didn't see any reference to setuptools in setup.{py,cfg} and pyproject.toml.

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-widgetsnbextension/package.py]: `setup.py` contains `from setuptools import setup`

**Review by adamjstewart — APPROVED** [review]

**Comment by iarspider** [comment]
@spackbot run pipeline

**Pull request merged**